### PR TITLE
Refactor template handler

### PR DESCRIPTION
**Pull request by alexharv074**
In 0abe66966bc1bd86df3884389a61edc675a8f98c, some unnecessary
refactoring occurred while fixing Issue #1148. This reverts that
refactoring and adds unit tests for the File handler.

## PR Checklist

- [X] Wrote a good commit message & description [see guide below].
- [ ] Commit message starts with `[Resolve #issue-number]`.
- [X] Added/Updated unit tests.
- [ ] Added/Updated integration tests (if applicable).
- [X] All unit tests (`make test`) are passing.
- [X] Used the same coding conventions as the rest of the project.
- [X] The new code passes pre-commit validations (`pre-commit run --all-files`).
- [X] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

**Review by zxiiro — APPROVED** [review]

File: sceptre/template_handlers/file.py [hunk 1]
```diff
@@ -50,16 +51,3 @@ def handle(self):
...
-        """
-        Return the project_path joined to template_path as
-        a string.
-
-        Note that os.path.join defers to an absolute path
-        if the input is absolute.
-        """
-        return path.join(
-            self.stack_group_config["project_path"], "templates",
-            normalise_path(template_path)
```
Comment (zaro0508) [inline sceptre/template_handlers/file.py:64]: wasn't this put in to make template handler work with windows paths?

Comment (alexharv074) [reply sceptre/template_handlers/file.py:64]: Perhaps @jfalkenstein could comment. At the time, I thought `normalise_path` was only required because I had moved some legacy code to the template handler class. Looking at it more closely, the new code may also have needed `normalise_path` to be applied to the `path` argument. What if I apply this patch here:

```diff
--- a/sceptre/template_handlers/file.py
+++ b/sceptre/template_handlers/file.py
@@ -4,7 +4,7 @@ import sceptre.template_handlers.helper as helper
 from pathlib import Path
 from sceptre.exceptions import UnsupportedTemplateFileTypeError
 from sceptre.template_handlers import TemplateHandler
-
+from sceptre.helpers import normalise_path
 
 class File(TemplateHandler):
     """
@@ -25,7 +25,7 @@ class File(TemplateHandler):
 
     def handle(self):
         project_path = self.stack_group_config.get("project_path")
-        input_path = Path(self.arguments["path"])
+        input_path = Path(normalise_path(self.arguments["path"]))
         if input_path.is_absolute():
             path = str(input_path)
         else:
```

Alternatively, I can just abort this part of the change and go with the version of the code that was already merged. However, I feel that I added unnecessary complexity.

Comment (jfalkenstein) [reply sceptre/template_handlers/file.py:64]: normalise_path takes a "sceptre_path" and turns it into an os-compatible path. This is important in the case of windows, where the user (per Sceptre's documentation) provides a template path like "directory/my-stack-config.yaml". Normalize_path will convert that to be windows-compatible. This is important for windows support.

Comment (jfalkenstein) [reply sceptre/template_handlers/file.py:64]: So yeah, I think we should probably still use it. Honestly... It's not clear to me why we're trying to put the code _back_ to it's previous state. The current state of this handler works well and accounts for all the edge cases we've discussed over the last week.

Comment (alexharv074) [reply sceptre/template_handlers/file.py:64]: @jfalkenstein Ok, how about we just add the tests on this branch.

**Review by zaro0508 — APPROVED** [review]
